### PR TITLE
vim-patch.sh: fix/improve pattern with find_git_remote

### PR DESCRIPTION
**Pull request by blueyed**
/cc @jamessan

### CODE REVIEW

File: scripts/vim-patch.sh [hunk 1]
```diff
@@ -87,7 +87,7 @@ commit_message() {
 }
 
 find_git_remote() {
-  git remote -v | awk '/neovim\/neovim.*(fetch)/{print $1}'
+  git remote -v | awk '$2 = /neovim\/neovim.*\(fetch\)$/ {print $1}'
```
Comment (jamessan) [inline scripts/vim-patch.sh]: What's the `$2 = ` for?  That doesn't restrict the scope of the match.  It just overwrites the value of `$2`.

If you're trying just to match in `$2`, you would need `$2 ~` but `(fetch)` isn't in `$2`.

Comment (blueyed) [inline scripts/vim-patch.sh]: Yeah, totally messed it up.  Should be fixed now.

**Comment by jamessan** [comment]
Was there a specific problem you encountered or is this just a preventative change?

File: scripts/vim-patch.sh [hunk 2]
```diff
@@ -87,7 +87,7 @@ commit_message() {
 }
 
 find_git_remote() {
-  git remote -v | awk '/neovim\/neovim.*(fetch)/{print $1}'
+  git remote -v | awk '$2 ~ /neovim\/neovim/ && $3 == "(fetch)" {print $1}'
```
Comment (jamessan) [inline scripts/vim-patch.sh]: Depending on how strict you want to be, it may be worth changing the first match to `$2 ~ /github.com[:/]neovim\/neovim/`.

Comment (blueyed) [inline scripts/vim-patch.sh]: Makes sense, done.

**Comment by blueyed** [comment]
It's just a preventative change.

**Pull request merged**